### ISSUE TITLE
openJ9 JDK11 with tomcat fails to propagate context

### ISSUE THREAD

**Issue by jackshirazi** [elastic/apm-agent-java#2203]
## Describe the bug
For both JakartaeeTomcatIT and TomcatIT classes in the integration test suite, OpenJ9 on JDK11 testing fails (classes have those configuration of tests commented out for now). The attached log shows the test run error, it appears to be because of JDK9+ modularization access changes

## Steps to reproduce
1. Uncomment any of the commented out lines in JakartaeeTomcatIT and TomcatIT classes in the integration test suite and run the test (eg lines shown here)
```
            // TODO openj9 on JDK11 has an access problem from java.base
            //{"10.0.10-jdk1
[error.txt](https://github.com/elastic/apm-agent-java/files/7373275/error.txt)
1-adoptopenjdk-openj9"},
```

## Expected behavior
Expect the test to pass with no failure, as per all the other tests in the same class (eg hotspot on jdk11 is fine)

## Debug logs
attached holds the output of the failed test run

**Comment by eyalkoren** [comment]
See #1415 and the fix we applied through #1468, look into why the same fix doesn't apply in this JVM.
Could be something stupid like [this check](https://github.com/elastic/apm-agent-java/pull/1468/files#diff-2ec63f545539ff7d65f52ef01087a18c7cdd9eb7130de156fb412ae6fe9e580eR216) failing.

**Comment by SylvainJuge** [comment]
The test still fails after fixing the module access issue with #2341
